tools/bin/hawqregister

```diff
@@ -325,250 +326,221 @@ class HawqRegister(object):
+    def check_no_regex_filepath(self, files):
+        for fn in files:
+            tmp_lst = fn.split('/')
+            for v in tmp_lst:
+                if v == '.':
+                    logger.error('Hawq register does not support file path with regex: %s.' % fn)
                     self.failure_handler.rollback()
                     sys.exit(1)
-            for k, fn in enumerate(self.files):
-                hdfscmd = 'hadoop fs -du %s' % fn
-                _, out, _ = local_ssh_output(hdfscmd)
-                if self.sizes[k] > int(out.strip().split()[0]):
-                    logger.error('File size(%s) in yaml configuration file should not exceed actual length(%s) of file %s.' % (self.sizes[k], out.strip().split()[0], fn))
+            for ch in ['..', '*']:
+                if fn.find(ch) != -1:
+                    logger.error('Hawq register does not support file path with regex: %s.' % fn)
                     self.failure_handler.rollback()
                     sys.exit(1)
 
-        def check_no_regex_filepath(files):
-            for fn in files:
-                tmp_lst = fn.split('/')
-                for v in tmp_lst:
-                    if v == '.':
-                        logger.error('Hawq register does not support file path with regex: %s.' % fn)
-                        self.failure_handler.rollback()
-                        sys.exit(1)
-                for ch in ['..', '*']:
-                    if fn.find(ch) != -1:
-                        logger.error('Hawq register does not support file path with regex: %s.' % fn)
-                        self.failure_handler.rollback()
-                        sys.exit(1)
-
+    def _init(self):
         if self.yml:
-            option_parser_yml(options.yml_config)
+            self.option_parser_yml(options.yml_config)
             self.filepath = self.files[0][:self.files[0].rfind('/')] if self.files else ''
-            check_file_not_folder()
-            check_database_encoding()
+            self.check_file_not_folder()
+            self.check_database_encoding()
             if self.mode != 'repair':
-                if not create_table() and self.mode != 'force':
+                if not self.create_table() and self.mode != 'force':
                     self.mode = 'usage2_table_exist'
-            check_bucket_number()
-            check_distribution_policy()
-            check_policy_consistency()
-            check_no_regex_filepath(self.files)
         else:
-            if is_folder(self.filepath) and self.filesize:
-                logger.error('-e option is only supported with single file case.')
-                sys.exit(1)
             self.file_format = 'Parquet'
-            check_hash_type() # Usage1 only support randomly distributed table
+            self.check_hash_type() # Usage1 only support randomly distributed table
+        self.do_check()
```

Comment:
The name here is a little confused I think.

Firstly, your L509 use `check_file_not_folder `, L510 `check_database_encoding `, L516 use `check_hash_type`, what's the difference between `check_file_not_folder`, `check_database_encoding`, `check_hash_type` and `do_check`. (I know the reason why you do this, but please refactor with semantic, not just code itself...)

Secondly, `do_check` is an internal interface used in `_init`, which means it can only be invoked in the right place of `_init` function.  I strongly recommend define this kind of function inside `_init`, it can not be invoked independently. Anther reason defining a function variable inside a function is simplify the invoking process for readability.

Thirdly, a internal interface in Python should be named starting with `_` sign.